.ci/validation.sh

```diff
@@ -669,9 +669,9 @@ no-error-checkstyles-sevntu)
   CS_POM_VERSION="$(getCheckstylePomVersion)"
   echo CS_version: "${CS_POM_VERSION}"
   mvn -e --no-transfer-progress clean install -Pno-validations
-  mvn -e --no-transfer-progress compile verify \
+  mvn -e --no-transfer-progress compile verify -Psevntu \
```

Comment:
Do you think jar usage is better than classpath from target folder?
Potentially we can skip extra build step and run it more easily on local.
Or it is good live example of ant usage of custom jar ?

Reply:
> classpath from target folder

I am unsure what this part of the statement means. Sevntu profile run is using officially released checkstyle version. It is not using the master version, which is what is stored in target. Only results and such are put into target folder.

> Potentially we can skip extra build step and run it more easily on local.
> Or it is good live example of ant usage of custom jar ?

If your referring to JAR usage, we technically have to package the JAR for us to use it. Also, if we switch to JAR usage, we have to then go through the trouble of downloading the sevntu dependency to add it and use it in the run. I think it is easier to maintain if we continue to rely on maven.

I also believe we added plenty of examples to https://github.com/sevntu-checkstyle/checkstyle-samples and can point people to that if they need different ways of running checkstyle.

Reply:
We build snapshot jar

And reference it at
`Dmaven.sevntu-checkstyle-check.checkstyle.version="${CS_POM_VERSION`
We did this as maven does not have ability to work on folder with classes.
As we use our own ant task, we can do same as main execution.

We want to test that code in PR is not making breaking changes to sevntu.

Reply:
@romani What is the change you want me to make? I am not understanding if it was stated.

Reply:
I still do not see reason to run sevntu as separate profile.
With new profile, we stik sevntu from "mvn verify"  - bad.

I do not see good reason to keep running on released version of checkstyle. We need to run on checkstyle target folder + sevntu jar.

We use sevntu as maven execution only as convenience, or my lack of knowledge at that time to run ANT smartly.
As we remove maven plugin,  sevntu should join our regular ant phase, it is what we need and it will be more beneficial.

So there will be no special profile, sevntu runs on freshly compiled sources.
It will allow us to remove special job in CI where we try to run sevntu on SNAPSHOT checkstyle jar.

Reply:
If we want to run Sevntu on an unsupported version of Checkstyle, then lets make this another PR with its own separate isolation.

But just to note, it is a bad idea to run Sevntu on unsupported versions of Checkstyle. When sevntu was stuck on 10.0, it didn't make sense to try and run it on 10.4, and if there was a breaking enough change, this means DISABLING the entire sevntu run, even though we could just run it against the supported 10.0 . Checking sevntu works with the latest CS version is a separate no-violation regression that isn't tested here.

Reply:
from rnveach:

my concern I brought up is we were breaking sevntu

Your saying you want to force sevntu to run on an unconfirmed version of CS by running on snapshot instead of release

we only use release version of sevntu, so I am less worried about sevntu doing a breaking change

The question basically comes down to, if CS starts doing breaking changes. Do you still want to run sevntu against our code base or not? If you do, we can't run sevntu with the snapshot version of CS.

Imagine we had to change the name of something common between the 2. CI will report the issue in CS snapshot. Sevntu which relies on the released version of CS, can't make a change for something that is breaking in snapshot. It will never compile since the change is only in snapshot.

There are things only we can find in CI as it is. Checker and ArchUnit is one. Pitest is a better example.

Reply:
I am ok to keep us less dependent on satellites as whole idea of migration from maven to ant to let us move forward easier with breaking changes and let satellites to catch up.